libsql/src/statement.rs

```diff
@@ -92,18 +93,37 @@ pub struct MappedRows<F> {
     map: F,
 }
 
-impl<F, T> Iterator for MappedRows<F>
+impl<F> MappedRows<F> {
+    pub fn new(rows: Rows, map: F) -> Self {
+        Self { rows, map }
+    }
+}
+
+#[cfg(feature = "core")]
+impl<F, T> futures::Stream for MappedRows<F>
 where
-    F: FnMut(Row) -> Result<T>,
+    F: FnMut(Row) -> Result<T> + Unpin,
 {
     type Item = Result<T>;
 
-    #[inline]
-    fn next(&mut self) -> Option<Self::Item> {
-        let map = &mut self.map;
-        self.rows
-            .next()
-            .transpose()
-            .map(|row_result| row_result.and_then(map))
+    fn poll_next(
+        mut self: std::pin::Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Option<Self::Item>> {
+        use futures::{ready, Future};
+
+        let mut rows = unsafe { self.as_mut().map_unchecked_mut(|pin| &mut pin.rows) };
+        let mut fut = Box::pin(rows.next());
```

Comment:
FYI can use this to avoid the allocations each time https://docs.rs/tokio-util/latest/tokio_util/sync/struct.ReusableBoxFuture.html